wp-content/plugins/automatic-translation/TranslationManager.php

```diff
@@ -136,6 +159,53 @@ private function should_create_automatic_translation($current_translation_id) {
 		}
 		return false;
 	}
+	
+	//check if translation of parent exists in target language code
+	private function translated_parent_exists( $post, $target_language_code ) {
+		//if page has post_status inherit, get post_parent
+		//we dont need the current translation. the initial translation suffices
+		$post = get_post($post->ID);
+		$post_main_id = $post->ID;
+		if( $post->post_status == 'inherit' ) {
+			$post_main_id = $this->get_post_parent ( $post->ID );
+		} 
+		//get post_parent if post_status is not inherit
+		$post_parent_id = $this->get_post_parent ( $post_main_id );
+				
+		if( $post_parent_id == 0 ) {
+			return true; //there is no parent that needs translation
+		}
+		
+		$post->ID = $post_parent_id;
+		$post = get_post($post->ID);
+		
+		if( $this->wpml_helper->get_translated_post_id($post, $target_language_code) == NULL ) {
+			return false; //translation of parent is missing
+		} else {
+			return $post->ID;
+		}
+	}
+	
+	private function get_post_parent ( $post_id ) {
+		global $wpdb;
+		
+		$query = "SELECT post_parent FROM $wpdb->posts WHERE ID = '$post_id' AND post_status = 'inherit'";
+		$parent = $wpdb->get_results($query, OBJECT);
+		if( $parent[0]->post_parent )
+			$post_id = $parent[0]->post_parent;
+		
+		$query = "SELECT post_parent FROM $wpdb->posts WHERE ID = '$post_id'";
+		$parent = $wpdb->get_results($query, OBJECT);
+		return $parent[0]->post_parent;
+	}
+	
+	//parameters: post object of child, wpml source language code and target language code
```

Comment:
there is no benefit from this comment.
Refer to [phpdoc's @param](https://manual.phpdoc.org/HTMLSmartyConverter/HandS/phpDocumentor/tutorial_tags.param.pkg.html)